src/irmin-pack/dict.ml

```diff
@@ -79,13 +61,23 @@ module Make (IO : IO.S) : S = struct
     in
     (aux [@tailcall]) (Hashtbl.length t.cache) 0
 
-  let sync_offset t =
-    let former_log_offset = IO.offset t.io in
-    let log_offset = IO.force_offset t.io in
-    if log_offset > former_log_offset then refill ~from:former_log_offset t
-
-  let sync t =
-    if IO.readonly t.io then sync_offset t
+  let sync_offset t ~force_refill =
+    if force_refill then (
```

Comment:
do we really need a `force_refill` here? Can't we just read the generation number and refill only if it has changed?

Reply:
It's to save up one read, which I'm not sure is a big optimisation. But `pack` also reads the generation from file, and as `pack.sync` calls `Dict.sync`, then `force_refill` prevents an extra read for `Dict.sync`. I think the header should be shared between the two files, pack and dict, for now each has its own header, but they should have the same generation.

Reply:
Added a commit to remove the `force_refill`.